src/renderable/animationsheet.js

```diff
@@ -143,7 +143,7 @@
                             "string parameters for addAnimation are not allowed for standard spritesheet based Texture"
                         );
                     } else {
-                        this.anim[name].frame[i] = this.textureAtlas[this.atlasIndices[index[i]]];
+                        this.anim[name].frame[i] = Object.assign(this.textureAtlas[this.atlasIndices[index[i]]], { delay: this.anim[name].animationspeed });
```

Comment:
First point,  i didn't implement it this way, as I find it odd that we would have the texture atlas object contain delay information, when that is purely about image coordinates & rotation.

Second point, agreed it should be split up on a couple lines :)